File: js&ts/30DaysOfJS/lc-2405.js
/**
 * @param {string} s
 * @return {number}
 */
var partitionString = function (s) {
  let dict = {};
  let res = 1;
  for (let i = 0; i < s.length; i++) {
    if (dict[s[i]] == true) {
      dict = {};
      res++;
    }
    dict[s[i]] = true;
  }
  return res;
};

File: golang/lc-2405.go
package main

func partitionString(s string) int {
	res := 1
	dict := make(map[byte]bool)
	for i := 0; i < len(s); i++ {
		if dict[s[i]] == true {
			dict = make(map[byte]bool)
			res++
		}
		dict[s[i]] = true
	}
	return res
}
